citest/json_predicate/indexed_predicate_result.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implements a composite ValuePredicateResult using an indexed array."""
+
+from .predicate import (
+    CloneableWithNewSource,
+    PredicateResult)
+
+class IndexedPredicateResult(PredicateResult, CloneableWithNewSource):
+  """Aggregates a list of predicate results into a single response.
+
+  Attributes:
+    pred: The ValuePredicate doing the aggregation.
+    results: The list of PredicateResponse instances being aggregated.
+  """
+
+  @property
+  def pred(self):
+    """The predicate used to collect the composite results."""
+    return self.__pred
+
+  @property
+  def results(self):
+    """The list of PredicateResult instances."""
+    return self.__results
+
+  def export_to_json_snapshot(self, snapshot, entity):
+    builder = snapshot.edge_builder
+    summary = builder.object_count_to_summary(
+        self.__results, subject='indexed composite result')
+    builder.make_mechanism(entity, 'Predicate', self.__pred)
```

Comment:
The naming here is too generic as a leaf implementation of an interface. Yes, the `export_to_json_snapshot` interface should be generic, but when you giving it a concrete implementation in a class, more information needs encoded in the variable naming. I don't really know which `entity` we're referring to here in the snapshot. This resource graph is incredibly generic, so it's difficult to wrap my head around what's occurring without more specificity and comments. We'll talk offline about this -- I need a better grasp about how these are created and consumed.